spec/truffle/interop/foreign_string_spec.rb

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+# 
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+require_relative '../../ruby/spec_helper'
+
+describe "Foreign strings" do
+  
+  it "are boxed" do
+    Truffle::Interop.boxed?(Truffle::Debug.foreign_string('test')).should be_true
+  end
+  
+  it "unbox to Ruby strings" do
```

Comment:
`unbox to Java strings` and this should use unbox_without_conversion

Reply:
Hmm I think the point is that you can use `UNBOX` on a foreign string, and you'll get a Ruby string. That's what a user case about. I'm not literally writing a spec for the `ForeignString` object, as that's just a mock. The point is a foreign string will `UNBOX` to a Ruby string in all cases where you use unboxing as normal.

Reply:
That's only true if one uses `Truffle::Interop.unbox` though, so this seems better suited in Truffle::Interop.unbox's spec.
If some code just sends UNBOX it will get a Java String.
And I think most usages of UNBOX are not manual calls like this one.

Reply:
Actually, just describing it as `unbox to Ruby strings with Truffle::Interop.unbox` would make sense to me.

Reply:
> That's only true if one uses Truffle::Interop.unbox ... If some code just sends UNBOX it will get a Java String.

How are you going to send UNBOX without using `Truffle::Interop.unbox` or `Truffle::Interop.unbox_without_conversion`?

> I think most usages of UNBOX are not manual calls like this one

What other uses of `UNBOX` are there?

Reply:
These are literally the only `UNBOX` in operations in the entire code base.

Reply:
> How are you going to send UNBOX without using Truffle::Interop.unbox or Truffle::Interop.unbox_without_conversion?

I meant in Java code. UNBOX is usually used on the receiving end of interop.
For instance we could try to unbox a foreign String if someone tries to call define_method(foreignString) {} or passes a foreign string as a READ/WRITE name).

Anyway, I'm happy with the spec if it clarifies this is due to `Truffle::Interop.unbox` and not the UNBOX message itself.

Reply:
> I meant in Java code

Right - this is documentation for how a Ruby person would see it.

> Anyway, I'm happy with the spec if

Ok I'll change to that.